src/unix/fs.c

```diff
@@ -766,6 +770,135 @@ static ssize_t uv__fs_write(uv_fs_t* req) {
   return r;
 }
 
+static ssize_t uv__fs_copyfile(uv_fs_t* req) {
+#if defined(__APPLE__) && !TARGET_OS_IPHONE
+  /* On macOS, use the native copyfile(3). */
+  copyfile_flags_t flags;
+
+  flags = COPYFILE_ALL;
+
+  if (req->flags & UV_FS_COPYFILE_EXCL)
+    flags = flags | COPYFILE_EXCL;
+
+  return copyfile(req->path, req->new_path, NULL, flags);
+#else
+  uv_fs_t fs_req;
+  uv_file src;
+  uv_file dst;
+  uv_buf_t buf;
+  char data[4096];
+  struct stat statsbuf;
+  int offset;
+  int num_read;
+  int num_written;
+  int dst_flags;
+  int result;
+  int err;
+
+  dst = -1;
+
+  /* Open the source file. */
+  src = open(req->path, O_RDONLY);
```

Comment:
Sorry, I just realized I'm dumb. Our open will take of CLOEXEC and friends, so better use that. Sorry again.